ansible_ai_connect/ai/api/formatter.py

```diff
@@ -368,15 +368,33 @@ def get_task_names_from_tasks(tasks):
     return names
 
 
-def restore_original_task_names(output_yaml, prompt):
+def restore_original_task_names(output_yaml, prompt, payload_context=""):
     if output_yaml and is_multi_task_prompt(prompt):
-        prompt_tasks = get_task_names_from_prompt(prompt)
-        matches = re.finditer(r"^- name:\s+(.*)", output_yaml, re.M)
-        for i, match in enumerate(matches):
+        full_yaml = payload_context + output_yaml
+        try:
+            payload_context_data = yaml.safe_load(payload_context)
+            full_data = yaml.safe_load(full_yaml)
+        except Exception as exc:
+            logger.exception(
+                f"Error while loading the result role/playbook YAML:{exc} "
+                f"for restoring the original task names"
+            )
+            return output_yaml
```

Comment:
Do we want a request to keep processing and potentially ultimately give the user a response when we weren't able to fix up the output? I'd think no. We know that currently it's going to fail in subsequent steps due to invalid yaml, but if the code around this function is restructured that wouldn't be the case. Should we raise this exception instead of just letting processing continue?

Reply:
I think we should let the process continue but log it as an error to be traced just like we discussed before